### PR TITLE
Jeffrey2

### PR DESCRIPTION
**Pull request by JeffTheAggie**
### Reference to issue
- #28 
- #59 

### Description of the changes proposed in the pull request
- Added Jupyter Notebooks that showcased the two examples of supervised learning from my "Intro to Supervised Learning" article through using Linear Regression and Logistic Regression (this was for the classification example)
- Forked branch from bitproject to my profile, made a pull request of forked branch to branch on repo, and now will submit a pull request from the updated branch to the master branch
### Reviewers requested:
- @your_PM_github_username (IDK yet because I never saw the assignment for that so I assume I'll send this to @shreyagupta98 )
- @your_QA_github_username (Same reason as previous bullet point)

### CODE REVIEW

File: Intro_to_Supervised_Learning.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Intro to Supervised Learning by, Jeffrey Ugochukwu
```
Comment (shreyagupta98) [inline Intro_to_Supervised_Learning.md:1]: You can remove your name here, it will appear in the author section.

File: Intro_to_Supervised_Learning.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Well the answer to that is based on these factors:
+- The size of the data would matter since a large sample size would lead to less variability compared to a small sample size
+- Distribution of data
+- The type of relationship between the labels and features of dataset (linear, non-linear, etc...)
+- Data format such as structured data (csv files, txt files, etc...) and unstructured data (json files, audio files, etc...)
+- Goal of the actual analysis itself
+
+## Accuracy vs Interpretation
+Now in terms of detecting the outcomes of our machine learning, we should take into account about the level of accuracy that our model creates and how much it inteprets what the true value should be. This is alo dependent on how flexible our model is, which means how much our fitted model varies from the data we have to train. The more flexible our model is, the less variation it has from the training data and it could fit more distributions of shapes towards the data, making it more accurate. The less flexible our model is, the more variation it has towards the data, which would fit less distributions of shapes towards the data, thus making it more interpretable
```
Comment (shreyagupta98) [inline Intro_to_Supervised_Learning.md:33]: Look out for typos "alo" > "also" and the repetition of words like "now".

File: Intro_to_Supervised_Learning.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Intro to Supervised Learning by, Jeffrey Ugochukwu
+
+## What is Supervised Learning?
+       
+Hello, fellow reader. I see that you're very interested in finding interesting applications of using Statistics to solve many
```
Comment (shreyagupta98) [inline Intro_to_Supervised_Learning.md:5]: Jeffrey, I like what you were trying to do with making this blog more personable for the reader. I would recommend removing phrases that directly target the reader like "Hello, fellow reader" and "you" (or lessing the number of times you use them).

File: Intro_to_Supervised_Learning.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+- Distribution of data
+- The type of relationship between the labels and features of dataset (linear, non-linear, etc...)
+- Data format such as structured data (csv files, txt files, etc...) and unstructured data (json files, audio files, etc...)
+- Goal of the actual analysis itself
+
+## Accuracy vs Interpretation
+Now in terms of detecting the outcomes of our machine learning, we should take into account about the level of accuracy that our model creates and how much it inteprets what the true value should be. This is alo dependent on how flexible our model is, which means how much our fitted model varies from the data we have to train. The more flexible our model is, the less variation it has from the training data and it could fit more distributions of shapes towards the data, making it more accurate. The less flexible our model is, the more variation it has towards the data, which would fit less distributions of shapes towards the data, thus making it more interpretable 
+(less accurate). 
+
+## Regression Example with Scikit-learn
```
Comment (shreyagupta98) [inline Intro_to_Supervised_Learning.md:36]: Is the rest of the blog going to be code alone or are you going to integrate code into the blog?

File: contentdevproposal.md [hunk 5]
```diff
@@ -2,26 +2,32 @@
...
 
 **Middle (2-3+ sentences):**    
-{{Insert_here}}
+This is where I'll give the examples of using Regression and Classification models. A major component of it is creating the code snippets and explaining how the written functions in those snippets are manipulating our data in some way. This is also where I explain the mathematics for each chosen model (explaing the fundamentals of the linear regression formula and how the Root Mean Square Error (RMSE) is the measure of how far off the trained data is from the correct value (for regression); will also talk about how logistic regression (a type of classification model) will represent how our data corresponds to a binary variable (i.e. yes/no, male/female, etc...) and also measure how true (meaning accurate) positive/negative and false (meaning predicted) positive/negative values are calculated as in 4 proportions).
```
Comment (shreyagupta98) [inline contentdevproposal.md:30]: Looking forward to seeing more on this in your blog!

**Comment by shreyagupta98** [comment]
@JeffTheAggie please make sure to follow the instruction of which branch to merge to (not master)

**Comment by JeffTheAggie** [comment]
I already finished writing the outline for my article. You can check the branch Jeffrey2 to see it

File: Intro_to_Supervised_Learning.md [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+- The size of the data would matter since a large sample size would lead to less variability compared to a small sample size
+- Distribution of data
+- The type of relationship between the labels and features of dataset (linear, non-linear, etc...)
+- Data format such as structured data (csv files, txt files, etc...) and unstructured data (json files, audio files, etc...)
+- Goal of the actual analysis itself
+
+## Accuracy vs Interpretation
+Now in terms of detecting the outcomes of our machine learning, we should take into account about the level of accuracy that our model creates and how much it inteprets what the true value should be. This is also dependent on how flexible our model is, which means how much our fitted model varies from the data we have to train. The more flexible our model is, the less variation it has from the training data and it could fit more distributions of shapes towards the data, making it more accurate. The less flexible our model is, the more variation it has towards the data, which would fit less distributions of shapes towards the data, thus making it more interpretable 
+(less accurate). 
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:35]: The first 4 sections are very thorough and you explained a lot. However for a blog, its a bit wordy. Can you shorten it to just the main intro ideas?

File: Intro_to_Supervised_Learning.md [hunk 7]
```diff
@@ -0,0 +1,525 @@
...
+- bmi: the patient's body max index
+- bp: average blood pressure
+- s1: T-Cells (a type of white blood cell)
+- s2: low density lipoproteins
+- s3: high density lipoproteins
+- s4: tch (thyroid stimulating hormone)
+- s5: lamotrigine (medication to can control blood pressure levels)
+- s6: blood sugar level
+
+Since the median blood pressure level isn't a given column in this dataset, we can create through expressing it this way.
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:103]: What is "median blood pressure level"? I see a column named "bp" already, the average. Whats the point of this? If the goal is to predict blood pressure, state it.

File: Intro_to_Supervised_Learning.md [hunk 8]
```diff
@@ -0,0 +1,525 @@
...
+
+Now we're going into the split the data for testing and training. The reason as to why we would seperate our data into two groups is because the testing data is new data that the model hasn't seen before so it would use that as a basis to predict the median blood pressure level whereas the training data is familiar to the model. This would defeat the purpose of this procedure since this would mean that our model will only make predictions based on what it knows rather then working with any form of data that's unfamiliar to it. To create the actual model, we first would need to create the x variable for fetaures (in this case, the last index of the entire dataset) and the y variable for the actual Blood Pressure column. Then we will call the `train_test_split()` to split our data for x and y for testing and training. We will set 80% of the data for training and 20% of it for testing (explains why `test_size = 0.2`). Now that the conceptual material has been explained, let's find the size of our testing and training data.
+
+```python
+#Setting up our data for training and testing
+X = diabetes_data.iloc[:,:-1]
+y = diabetes_data["Blood Pressure"]
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:128]: Awesome concept explanation. But I dont see much code description. What does ".iloc" mean?

File: Intro_to_Supervised_Learning.md [hunk 9]
```diff
@@ -0,0 +1,525 @@
...
+![](https://cdn-images-1.medium.com/fit/t/1600/480/1*a0hsu_ldXOKOq88Re4xYtQ.png)
+
+As you can see by this image that it's a summation of the linear regression function where each Xi is a different feature that's given in your dataset. The minimum RSS is achieved when the values of the true and predicted labels are very close which results in the creation of the Bi's.
+
+To measure the accuracy of our model, we will use the Root Mean Square Error (RMSE) formula which is the square root of the RSS of each data point. The formula can be expressed as this:
+
+![](https://secureservercdn.net/160.153.137.16/70j.58d.myftpupload.com/wp-content/uploads/2019/03/rmse-2.png)
+
+Now that we have gotten over the important mathematical concepts to build the regression model, let's put this knowledge into practice. The first step is to import the built in Diabetes dataset from sckiit-learn.
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:54]: import all packages first then get to coding

File: Intro_to_Supervised_Learning.md [hunk 10]
```diff
@@ -0,0 +1,525 @@
...
+#Using Diabetes dataset to predict the median blood pressure level for diabetes patients
+from sklearn.datasets import load_diabetes
+diabetes = load_diabetes()
+```
+
+The next step would be to convert the dataframe into a pandas dataframe to create the dataset into tabular form and will define our column names as the feature names of the built in dataset.
+
+```python
+#Converting dataset to Pandas Dataframe
+import pandas as pd
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:65]: what is pandas?

File: Intro_to_Supervised_Learning.md [hunk 11]
```diff
@@ -0,0 +1,525 @@
...
+X = diabetes_data.iloc[:,:-1]
+y = diabetes_data["Blood Pressure"]
+
+from sklearn.model_selection import train_test_split
+#Split the data into 80% training and 20% testing. 
+#The random_state allows us to make the same random split every time.
+X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2,random_state=327)
+print('Training data size: (%i,%i)' % X_train.shape)
+print('Testing data size: (%i,%i)' % X_test.shape)
+```
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:135]: whats sklearn?

File: Intro_to_Supervised_Learning.md [hunk 12]
```diff
@@ -0,0 +1,525 @@
...
+plt.xlabel('Actual Blood Pressure')
+plt.ylabel('Predicted Blood Pressure')
+plt.show()
+```
+
+![](https://www.picturepaste.ca/images/2020/05/23/Blood-Pressure-Regression-Line.png)
+
+As we can see here, in the lower end of the line of best fit, the model tends to overpredict what the median blood pressure level is (most of the data points are above the line of best fit) and at the higher end, the model tends to underpredict what the median blood pressure level is (most of the data points are below the best line of fit). Overall, our model could have done a much better job in terms of making the predictions much closer to the true median blood pressure level, but at least you're now aware about what factors to take account for when you're creating you're linear regression model
+
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:239]: this paragraph concludes what Linear regression is well. Lets stop here. Save the rest of the classification for another blog.

File: Intro_to_Supervised_Learning.md [hunk 13]
```diff
@@ -0,0 +1,525 @@
...
+```python
+Test RMSE: 5527.810024
+```
+
+The RMSE score means that the error in the predicted median blood pressure level is by approximately 5528 mmHg. Given that the range of blood pressure levels is between 25000 mmHg and 34600 mmHg this is an ordinary difference. The reason as to what may cause this difference in the first place might be that either the relationship between the features and the predicted values aren't perfectly linear or some of the features don't correlate very well with the median blood pressure level. To truly figure out where our model stands in terms of fitting the best line of fit, we shall plot the model as shown below. 
+
+```python
+#Predicted vs Actual Median Blood Pressure level for Diabetes dataset
+import matplotlib.pyplot as plt
+import seaborn as sns
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:226]: imports at the very top before code

File: Intro_to_Supervised_Learning.md [hunk 14]
```diff
@@ -0,0 +1,525 @@
...
+
+Now we're going to test the model on new data. We do this by creating a variable called `y_pred` which contains the predicted median Blood Pressure levels and we use this against `y_test` which contains the true Blood Pressure levels. We would also need to calculate the RMSE to see how much our model varies from the true median Blood Pressure level.
+
+```python
+#Testing the validity of the performance of our model by seeing how far off it is from the true  median Blood Pressure level
+from sklearn.metrics import mean_squared_error
+import numpy as np
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:211]: import to the top

File: Intro_to_Supervised_Learning.md [hunk 15]
```diff
@@ -0,0 +1,525 @@
+# Intro to Supervised Learning
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:2]: lets focus on Regression only as the intro to supervised learning. I would call it" Intro to Regression Analysis: Predicting blood pressure" Why? Because you provided both code and analysis of the output

File: Intro_to_Supervised_Learning.md [hunk 16]
```diff
@@ -0,0 +1,525 @@
...
+```
+
+The next step would be to check the shape of the dataframe to see how large it is.
+
+```python
+#Seeing the shape of the dataframe
+diabetes_data.shape
+```
+
+```python
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:76]: this shape output can be uploaded as a screenshot instead of a code snippet

File: Intro_to_Supervised_Learning.md [hunk 17]
```diff
@@ -0,0 +1,525 @@
...
+(442, 10)
+```
+
+We can now see here that there are 442 patients in totatl with 10 specific features.
+
+Let's look at the first 5 rows to see what type of features that we're dealing with.
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:83]: by default, pandas.head() prints first 5 rows. make this clarification too. because if i wanted to see 10 rows its...head(10),

File: Intro_to_Supervised_Learning.md [hunk 18]
```diff
@@ -0,0 +1,525 @@
...
+diabetes_data["Blood Pressure"] = diabetes.target*100
+```
+
+Now we're going to add the column in our given dataset and also calculate the summary statistics to measure the percentiles of each category.
+
+```python
+#Adding the new column into the original data set and looking at the summary statistics
+diabetes_data.describe().transpose()
+```
+
```
Comment (chriztopherton) [inline Intro_to_Supervised_Learning.md:116]: why transpose?
